src/features/slackapi/slackapi-utils.ts

```diff
@@ -141,31 +188,46 @@ Have a great week!
    * @param timeRegistrations all time registrations after yesterday
    * @param previousWorkDays dates and the number of today
    */
-  export const postWeeklyMessage = (
+  export const postWeeklyMessage = async (
```

Comment:
Some major changes:

- Change function name to describe more accurately what the function actually does
- Change the return type to return list of objects that contain sent message and corresponding response. This makes it easier to handle errors and sent message.
- Change map to for loop which makes it easier to construct the data and makes logic more readable.
- Responses should be handled outside this function

Similar changes should be implemented to postDailyMessage function

```
  /**
   * Post a weekly summary slack message to users
   *
   * @param weeklyCombinedData list of combined timebank and slack user data
   * @param nonProjectTimes all non project times
   * @param timeRegistrations all time registrations after yesterday
   * @param previousWorkDays dates and the number of today
   */
  export const postWeeklyMessageToUsers = async (
    weeklyCombinedData: WeeklyCombinedData[],
    timeRegistrations:TimeRegistrations[],
    previousWorkDays: PreviousWorkdayDates,
    nonProjectTimes: NonProjectTime[]
  ): Promise<WeeklyMessageResult[]> => {
    const { weekStartDate, weekEndDate } = TimeUtilities.lastWeekDateProvider();
    const { yesterday, today } = previousWorkDays;

    const messageResults: WeeklyMessageResult[] = [];

    for (const userData of weeklyCombinedData) {
      const { slackId, personId, expected } = userData;

      const isAway = TimeUtilities.checkIfUserIsAwayOrIsItFirstDayBack(timeRegistrations, personId, expected, today, nonProjectTimes);
      const firstDayBack = TimeUtilities.checkIfUserIsAwayOrIsItFirstDayBack(timeRegistrations, personId, expected, yesterday, nonProjectTimes);

      const message = constructWeeklySummaryMessage(userData, weekStartDate.toISODate(), weekEndDate.toISODate());

      if (!isAway && !firstDayBack) {
        messageResults.push({
          message: message,
          response: await sendMessage(slackId, message.message)
        });
      }
    }

    return messageResults;
  };

  export interface WeeklyMessageResult {
    message: WeeklyMessageData;
    response: ChatPostMessageResponse;
  }
```